src/main/java/uk/nhs/hee/trainee/details/api/TraineeProfileResource.java

```diff
@@ -120,21 +122,25 @@ public ResponseEntity<Void> deleteTraineeProfile(@PathVariable String tisId) {
   /**
    * Verify the trainee against the provided details.
    *
-   * @param email    The email to match.
-   * @param gmc      The GMC number to match.
-   * @param postcode The post code to match.
+   * @param email The email to match.
+   * @param gmc   The GMC number to match.
+   * @param dob   The date of birth to match.
    * @return The matching trainee ID, or 404 if not verified or not unique
    */
   @GetMapping("/trainee-verify")
   public ResponseEntity<String> getVerifiedTraineeIds(@NotNull @RequestParam String email,
-                                                      @NotNull @RequestParam String gmc,
-                                                      @NotNull @RequestParam String postcode) {
-    List<String> traineeIds = service.getTraineeTisIdsByEmailGmcAndPostcode(email, gmc, postcode);
+      @NotNull @RequestParam String gmc,
+      @NotNull @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate dob) {
+    log.info("Request to verify trainee ID matching email '{}', GMC '{}' and DOB '{}'", email, gmc,
+        dob.toString());
```

Comment:
There are no log-injection safeguards here - given its an internally facing endpoint and none of the other endpoints have any safeguards either, this felt ok, but just flagging-it up.